plugin/storage/badger/spanstore/reader.go

```diff
@@ -346,53 +345,60 @@ func (r *TraceReader) durationQueries(query *spanstore.TraceQueryParameters, ids
 	return ids
 }
 
+func mergeJoinIds(left, right [][]byte) [][]byte {
```

Comment:
rename to `mergeEqualIds` ?

Reply:
Are the ids sorted? Maybe that should be documented somewhere.

Reply:
It's mentioned at the beginning of the package. Everything is sorted (it's a sorted K/V).

Reply:
As for the name, it's because the algorithm is called "sort-merge join" and is used in relational databases. Here the sorting phase happens in the DB and the merge phase in this code. It's pretty descriptive in my opinion since if someone wants to improve this method such as doing it parallel or using sharding from multiple badgers there are known algorithms for those variations too (which would underneath use this in any case).

Reply:
Thanks for the explanation. Now it rings a bell..